resources/web/docs_js/actions/modal.js

```diff
@@ -0,0 +1,16 @@
+const SET_MODAL = "SET_MODAL_VALUE";
+
+export const setModal   = val => ({type: SET_MODAL, value: val});
+export const openModal  = setModal.bind(null, true);
+export const closeModal = setModal.bind(null, false);
+
+const initialState = {isOpen: false};
+
+export const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SET_MODAL:
+      return {isOpen: action.value};
+    default:
+      return state;
+  }
+};
```

Comment:
Optional, but I've been finding that [`redux-actions`](https://redux-actions.js.org/) has been helping cut down on Redux action/reducer boilerplate quite a bit.

It's also helped me, in the Docker UI work, to start organizing my store by application subdomain rather than a file hierarchy based on Redux concepts. [Here's how my store is organized.](https://github.com/elastic/infra/tree/master/docker/container-library/ui/src/store/)

Not asking you to make the change, just a suggestion if/when this starts getting complex.

Reply:
Yea I thought about it and I do like redux-actions and how you used it.

> organizing my store by application subdomain rather than a file hierarchy based on Redux concepts

I think we basically have the same thing. Our actions files export a reducer, and the names of the files correspond to the key used in the store. I'm missing some sugar from redux-actions inside my files though.